pycalphad/core/equilibrium.py

```diff
@@ -79,7 +81,7 @@ def _eqcalculate(dbf, comps, phases, conditions, output, data=None, per_phase=Fa
     """
     if data is None:
         data = equilibrium(dbf, comps, phases, conditions, to_xarray=False)
-    active_phases = unpack_phases(phases) or sorted(dbf.phases.keys())
+    active_phases = unpack_phases(phases)
```

Comment:
Should this also be passed through `filter_phases`?

Reply:
This is already taken care of by the caller. Including extra phases won't affect the result here; in fact, the current implementation (before this fix) is falling back to setting this equal to _all_ the phases in the Database.

Reply:
If we assume that filtering is done by the caller, should this also assume that the phases are in canonical form and don't require unpacking?

Reply:
Right now a dict is what's passed by the caller:
`active_phases = {name: dbf.phases[name] for name in filter_phases(dbf, comps, phases)}`
I think it's more robust to leave it as-is.